src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/domain/handlers/presubmit/PaymentAppealPreparer.java

```diff
@@ -133,9 +143,23 @@ public PreSubmitCallbackResponse<AsylumCase> handle(
             response.addErrors(Collections.singleton("Cannot retrieve the fee from fees-register."));
             return response;
         }
+
+        if (isWaysToPay(callbackStage, callback, isLegalRepJourney(asylumCase))) {
+            try {
+                serviceRequestService.createServiceRequest(callback, fee);
+            } catch (Exception e) {
+                response.addErrors(Collections.singleton("Cannot create Service Request."));
+                return response;
+            }
+        }
         asylumCase.write(PAYMENT_STATUS, PAYMENT_PENDING);
 
         return new PreSubmitCallbackResponse<>(asylumCase);
     }
 
+    private boolean isLegalRepJourney(AsylumCase asylumCase) {
+        return asylumCase.read(JOURNEY_TYPE, JourneyType.class)
+            .map(journey -> journey == JourneyType.REP)
```

Comment:
Bit confused, so JourneyType.REP is Legal Rep, or not. If this is legal rep then perhaps orElse will be false?

Reply:
This is the same logic that used to be in the `canHandle` methods, I just extrapolated it into a method, so I could use it inside the `isWaysToPay` method. I think it defaults to true.

But it made me realize this should be called when the appeal type is EA, HU or PA, so I added this check too.